### ISSUE TITLE
Integrity error during social login

### ISSUE THREAD

**Issue by periwinkleFTW** [pennersr/django-allauth#2844]
#####################
UPDATE: I found the issue. Eventbrite API does not have an attribute `primary` for emails as per eventbrite API reference https://www.eventbrite.com/platform/api#/reference/user/retrieve-information-about-a-user-account/retrieve-information-about-a-user-account
But allauth tries to get the primary attribute and gets NULL in return
Here is the traceback that causes errors:
```

params | (UUID('e45893c6-b48b-4546-ab5e-b32c93472c58'),  'myemail@gmail.com',  None,  True)
-- | --
self | <django.db.backends.postgresql.base.CursorDebugWrapper object at 0x7f71d0801df0>
sql | ('INSERT INTO "account_emailaddress" ("user_id", "email", "verified", '  '"primary") VALUES (%s, %s, %s, %s) RETURNING "account_emailaddress"."id"')
```

######################

I am trying to implement a social signup/login but I keep getting integrity error. This [issue](https://github.com/pennersr/django-allauth/issues/259) had a similar error so I tried adding a `save` method but it did not help. 

The weird part is that I am getting `NULL` instead of boolean during the process

Here is the error:
```

Environment:


Request Method: GET
Request URL: http://127.0.0.1:8600/accounts/eventbrite/login/callback/?state=6qiTCuaMbxNy&code=ECPIKBQIUOD5VEQUGGJR

Django Version: 3.1.7
Python Version: 3.8.9
Installed Applications:
['django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.sites',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'django.contrib.redirects',
 'django_extensions',
 'rest_framework',
 'storages',
 'corsheaders',
 'djangoql',
 'post_office',
 'allauth',
 'allauth.account',
 'allauth.socialaccount',
 'allauth.socialaccount.providers.eventbrite',
 'crispy_forms',
 'usermodel.apps.UsermodelConfig',
 'pages.apps.PagesConfig',
 'tickets.apps.TicketsConfig']
Installed Middleware:
['corsheaders.middleware.CorsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'whitenoise.middleware.WhiteNoiseMiddleware',
 'django.contrib.redirects.middleware.RedirectFallbackMiddleware',
 'django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware']



Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)

The above exception (null value in column "verified" of relation "account_emailaddress" violates not-null constraint
DETAIL:  Failing row contains (12,myemail@gmail.com, null, t, e7f396af-a013-43fe-ae8c-a0d17c0ae76c).
) was the direct cause of the following exception:
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.8/contextlib.py", line 75, in inner
    return func(*args, **kwds)
  File "/usr/local/lib/python3.8/site-packages/allauth/socialaccount/providers/oauth2/views.py", line 77, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/allauth/socialaccount/providers/oauth2/views.py", line 147, in dispatch
    return complete_social_login(request, login)
  File "/usr/local/lib/python3.8/site-packages/allauth/socialaccount/helpers.py", line 151, in complete_social_login
    return _complete_social_login(request, sociallogin)
  File "/usr/local/lib/python3.8/site-packages/allauth/socialaccount/helpers.py", line 172, in _complete_social_login
    ret = _process_signup(request, sociallogin)
  File "/usr/local/lib/python3.8/site-packages/allauth/socialaccount/helpers.py", line 42, in _process_signup
    get_adapter(request).save_user(request, sociallogin, form=None)
  File "/usr/local/lib/python3.8/site-packages/allauth/socialaccount/adapter.py", line 84, in save_user
    sociallogin.save(request)
  File "/usr/local/lib/python3.8/site-packages/allauth/socialaccount/models.py", line 249, in save
    setup_user_email(request, user, self.email_addresses)
  File "/usr/local/lib/python3.8/site-packages/allauth/account/utils.py", line 308, in setup_user_email
    a.save()
  File "/usr/local/lib/python3.8/site-packages/django/db/models/base.py", line 753, in save
    self.save_base(using=using, force_insert=force_insert,
  File "/usr/local/lib/python3.8/site-packages/django/db/models/base.py", line 790, in save_base
    updated = self._save_table(
  File "/usr/local/lib/python3.8/site-packages/django/db/models/base.py", line 895, in _save_table
    results = self._do_insert(cls._base_manager, using, fields, returning_fields, raw)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/base.py", line 933, in _do_insert
    return manager._insert(
  File "/usr/local/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/query.py", line 1254, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
  File "/usr/local/lib/python3.8/site-packages/django/db/models/sql/compiler.py", line 1397, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 98, in execute
    return super().execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 66, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 75, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.8/site-packages/django/db/utils.py", line 90, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.8/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)

Exception Type: IntegrityError at /accounts/eventbrite/login/callback/
Exception Value: null value in column "verified" of relation "account_emailaddress" violates not-null constraint
DETAIL:  Failing row contains (12, amyemail@gmail.com, null, t, e7f396af-a013-43fe-ae8c-a0d17c0ae76c).



```

I have also tried disabling  email verification for social login, but no luck:
```
ACCOUNT_EMAIL_VERIFICATION = 'mandatory'
SOCIALACCOUNT_EMAIL_VERIFICATION = 'none'
```

Here are my custom signup and login forms:
```
class CustomSignupForm(SignupForm):
    first_name = forms.CharField(max_length=30,
                                 widget=forms.TextInput(attrs={'placeholder': 'First name'}))
    last_name = forms.CharField(max_length=30,
                                widget=forms.TextInput(attrs={'placeholder': 'Last name'}))

    def save(self, user):
        user.first_name = self.cleaned_data['first_name']
        user.last_name = self.cleaned_data['last_name']
        user.save()
        return user

    def __init__(self, *args, **kwargs):
        super(CustomSignupForm, self).__init__(*args, **kwargs)

        self.fields["email"].label = ""
        self.fields["first_name"].label = ""
        self.fields["last_name"].label = ""
        self.fields["password1"].label = ""
        self.fields["password2"].label = ""



class CustomLoginForm(LoginForm):
    def __init__(self, *args, **kwargs):
        super(CustomLoginForm, self).__init__(*args, **kwargs)

        self.fields["login"].label = ""
        self.fields["password"].label = ""
```

**Comment by pennersr** [comment]
Closed via 48c34d68

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/2844
